### PR TITLE
libgd: Only add getopt-for-visual-studio if compiler is visual studio

### LINKED ISSUE #12069: [package] Libgd build breaks when with_png=true on linux
# Libgd build breaks when with_png=true on linux

## Problem description
It seems like the build definition is unconditionally depending on `getopt-for-visual-studio/20200201` if `with_png` is enabled:
https://github.com/conan-io/conan-center-index/blob/7c3f81b16596a01e9d4785d98f996dd367901524/recipes/libgd/all/conanfile.py#L58

Looking at the recipe for `getopt-for-visual-studio`, it seems to just raise an exception if the compiler is not Visual Studio:
https://github.com/conan-io/conan-center-index/blob/7c3f81b16596a01e9d4785d98f996dd367901524/recipes/getopt-for-visual-studio/all/conanfile.py#L17-L18

Thus, I think a simple fix to the problem is to only add this dependency in the libgd recipe with the inverse condition of configure function in getopt-for-visual-studio:
```python
    def requirements(self):
        ...
        if self.options.with_png:
            self.requires("libpng/1.6.37")
            if self.settings.compiler == "Visual Studio":
                self.requires("getopt-for-visual-studio/20200201")
        ...
```

## System information
Conan version: 1.51.0
Operating system: Ubuntu 18.04 running on WSL
Compiler: GCC 7.4.0

## How to reproduce (in principle):
- Use any profile that's not visual studio, and a clean conan index.
- Run `conan install libgd/2.3.2@ -o libgd:with_png=True --build=missing`

You should then see the following error:
```
ERROR: getopt-for-visual-studio/20200201: Invalid configuration: getopt-for-visual-studio is only supported for Visual Studio
```

I expect that this error should not show up, as I'm not on a Visual Studio compiler.

### PR DESCRIPTION
**Pull request by ulrichji**
fixes #12069

The occurred when building the libgd package having with_png=true.
This would add a dependency to getopt-for-visual-studio which
is configured to fail immediately if the compiler is not visual studio.
The CI was probably able to build this since the with_png option is by
default false.

The proposed solution is to only add getopt-for-visual-studio if the
detected compiler is in fact visual studio, and only then.

Specify library name and version:  **libgd/2.3.2**

I am submitting this PR because I use libgd in my project, but wasn't able to build it for linux when png was enabled. Investigating the recipe, I discovered that the problem is a bug in the recipe, and I would like to contribute a fix to the community.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libgd/all/conanfile.py [hunk 1]
```diff
@@ -55,7 +55,8 @@ def requirements(self):
         self.requires("zlib/1.2.11")
         if self.options.with_png:
             self.requires("libpng/1.6.37")
-            self.requires("getopt-for-visual-studio/20200201")
+            if self.settings.compiler == "Visual Studio":
```
Comment (uilianries) [inline recipes/libgd/all/conanfile.py:58]:
```suggestion
            if is_msvc(self):
```
from conan.tools.microsoft import is_msvc

Comment (ulrichji) [reply recipes/libgd/all/conanfile.py]: Thanks for the suggestion! However, `is_msvc` was added in [1.45.0](https://github.com/conan-io/conan/releases/tag/1.45.0), and the current conanfile specifies the following: `required_conan_version = ">=1.33.0"`, so I will also update that line.

File: recipes/libgd/all/conanfile.py [hunk 2]
```diff
@@ -55,7 +56,8 @@ def requirements(self):
         self.requires("zlib/1.2.11")
```
Comment (uilianries) [inline recipes/libgd/all/conanfile.py:56]:
```suggestion
        self.requires("zlib/1.2.12")
```

**Review by uilianries — COMMENTED** [review]
- Please, update dependencies version too:
  - zlib/1.2.11 -> zlib/1.2.12
  - libtiff/4.3.0 -> libtiff/4.4.0
  - freetype/2.11.0 -> freetype/2.12.1

- You need to update some imports  due Conan v2 migration, the Conan v2 linter shows what should be updated: https://github.com/conan-io/conan-center-index/blob/master/docs/v2_linter.md#import-tools-from-conan

**Review by uilianries — APPROVED** [review]
LGTM

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**